src/probnum/quad/_bayesquad.py

```diff
@@ -107,25 +116,123 @@ def bayesquad(
     >>> print(F.mean)
     0.5000
     """
+
+    # Error management
     if domain is None and measure is None:
         raise ValueError(
             "You need to either specify an integration domain or an integration "
             "measure. The Lebesgue measure can only operate on a finite domain."
         )
 
-    # Integration measure
-    if measure is None:
-        measure = LebesgueMeasure(domain=domain, dim=input_dim)
-
-    # Choose Method
-    bq_method = BayesianQuadrature.from_interface(
-        input_dim=input_dim, kernel=kernel, method=method, policy=policy
+    if domain is not None:
+        if isinstance(measure, GaussianMeasure):
+            raise ValueError("GaussianMeasure cannot be used with finite bounds.")
+        elif isinstance(measure, LebesgueMeasure):
+            warnings.warn(
+                "Both domain and a LebesgueMeasure are specified. The domain information will be ignored."
+            )
+
+    bq_method = BayesianQuadrature.from_bayesquad(
+        input_dim=input_dim,
+        kernel=kernel,
+        measure=measure,
+        domain=domain,
+        policy=policy,
+        max_nevals=max_nevals,
+        var_tol=var_tol,
+        rel_tol=rel_tol,
+        batch_size=batch_size,
     )
 
-    if nevals is None:
-        nevals = input_dim * 25
+    # Integrate
+    integral_belief, bq_state = bq_method.integrate(fun=fun)
+
+    return integral_belief, bq_state.info
+
+
+def bayesquad_fixed(
```

Comment:
What's the reasoning behind having a second interface here instead of adding optional arguments `nodes` and `fun_evals` in `bayesquad`? Potentially with a warning if a policy other than the default is specified and nodes and function evaluations are given.

Reply:
Yeah, that;s a good question, and it;s good you raised it. From my current point of view, `bayesquad` and `bayesquad_fixed` are two quite different methods. One runs an acquisition loop and actively selects data. so you provide the problem (callable) in the API. The other one does not acquire data, so you request not the problem but the data in the API (basically what `bayesquad` *is*, you request there). I find convoluting those to different problems settings into one API more confusing than helpful. You;d probably have to read the docs several minutes and then still be puzzled (but perhaps I just get puzzled easily :)) Scipy also has several APIs for quadrature methods that differ for this reason. I;d be interested to hear more opinions on this though @alpiges and @tskarvone .

Reply:
Great you bring this up---we've had extended conversations about this. We went for this setup mostly for the reasons given by @mmahsereci. In `scipy.integrate` you need to use a spline method such as `simpson` or `romb` to use on a data array.
Integrating this method into `bayesquad` might make the interface more confusing when all parameters are optional, but either `fun` or `fun-evals` must be given, and if `fun_evals` is given you also need nodes etc.

However, we might consider a merger in the future, in case the user already has some `fun_evals`, but wants to use BQ to collect some more. But that's too niche for the instant.

(on a different note: What might be confusing is that `fixed_quad` in `scipy` denoted fixed order Gaussian quadrature. Maybe we want to rename our method not to be mistaken for a probabilistic version of `fixed_quad`. Any ideas?)

Reply:
How about `bayesquad_from_data`?

Reply:
Thanks for the clarification. I understand the difference now. I like `bayesquad_from_data`. `bayesquad_fixed` sounds a little like the method has been fixed (as in it was broken before). So really this is a convenient interface for the constructor `BayesianQuadrature.from_data()`, correct?

Reply:
Currently there is no separate constructor for the `from_data` version in `BayesianQuadrature` (see [here](https://github.com/probabilistic-numerics/probnum/pull/420#discussion_r657021351)), because the constructor is problem-agnostic at this point. This is something we could discuss.
Both `bayesquad` and `bayesquad_from_data` therefore use the same constructor.